discovery/manager.go

```diff
@@ -387,7 +387,12 @@ func (m *Manager) updateGroup(poolKey poolKey, tgs []*targetgroup.Group) {
 	}
 	for _, tg := range tgs {
 		if tg != nil { // Some Discoverers send nil target group so need to check for it to avoid panics.
-			m.targets[poolKey][tg.Source] = tg
+			// Remove the deleted target.
+			if len(tg.Targets) == 0 && len(tg.Labels) == 0 {
```

Comment:
```suggestion
			if tg.Delete == true {
```

A small suggestion, you can refer to it to see if it is helpful to you.

Is it safer to describe the optimization of pod deletion through the displayed Delete tag instead of destroying the original test? I saw that this test is applied to the function of ApplyConfig, not the link of scheduled task update. That is to say, Now this way of writing has too much influence.

```go
in pod.go file ,we can add Delete: true
		send(ctx, ch, &targetgroup.Group{Source: podSourceFromNamespaceAndName(namespace, name), Delete: true})
```